expression/builtin_cast_vec.go

```diff
@@ -118,3 +118,38 @@ func (b *builtinCastIntAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.
 func (b *builtinCastIntAsRealSig) vectorized() bool {
 	return true
 }
+
+func (b *builtinCastTimeAsDurationSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
```

Comment:
Why do we need to implement vectorized evaluation for `builtinCastTimeAsDurationSig` in this PR?

Reply:
Here's an example.
```mysql
create table str(t varchar(20));
insert into str value ('1970-01-01');
insert into str value ('1970-01-01');

select addtime(cast(t as date), '3') from str;
+-------------------------------+
| addtime(cast(t as date), '3') |
+-------------------------------+
| 00:00:03                      |
| 00:00:03                      |
+-------------------------------+
```

and the expression tree of argument 0 is 
`arg0 -> builtinCastTimeAsDurationSig -> builtinCastStringAsTimeSig`

It will hit this line https://github.com/pingcap/tidb/blob/165d5ec1310d4d098b4ea50962826fa076b7e624/expression/builtin.go#L106